connor/traderModule.go

```diff
@@ -0,0 +1,539 @@
+
+	hashingPower     = 1
+	costPerkWh       = 0.0
+	powerConsumption = 0.0
+)
+
+type TraderModule struct {
+	c      *Connor
+	pool   *PoolModule
+	profit *ProfitableModule
+}
+
+func NewTraderModules(c *Connor, pool *PoolModule, profit *ProfitableModule) *TraderModule {
+	return &TraderModule{
+		c:      c,
+		pool:   pool,
+		profit: profit,
+	}
+}
+
+type DeployStatus int32
+
+const (
+	DeployStatusDEPLOYED    DeployStatus = 3
+	DeployStatusNOTDEPLOYED DeployStatus = 4
+)
+
+type OrderStatus int32
+
+const (
+	OrderStatusCancelled OrderStatus = 3
+	OrderStatusReinvoice OrderStatus = 4
+)
+
+func (t *TraderModule) getTokenConfiguration(symbol string, cfg *Config) (float64, float64, float64, error) {
+	switch symbol {
+	case "ETH":
+		return cfg.ChargeIntervalETH.Start, cfg.ChargeIntervalETH.Destination, cfg.Distances.StepForETH, nil
+	case "ZEC":
+		return cfg.ChargeIntervalZEC.Start, cfg.ChargeIntervalZEC.Destination, cfg.Distances.StepForZEC, nil
+	case "XMR":
+		return cfg.ChargeIntervalXMR.Start, cfg.ChargeIntervalXMR.Destination, cfg.Distances.StepForXMR, nil
+	}
+	return 0, 0, 0, nil
+}
+
+func (t *TraderModule) ChargeOrdersOnce(ctx context.Context, symbol string, token watchers.TokenWatcher, snm watchers.PriceWatcher, balanceReply *sonm.BalanceReply) error {
+	t.c.db.CreateOrderDB()
+	start, destination, step, err := t.getTokenConfiguration(symbol, t.c.cfg)
+	if err != nil {
+		log.Printf("cannot get token configuraton %v", err)
+		return err
+	}
+
+	count, err := t.c.db.GetCountFromDB()
+	if err != nil {
+		log.Printf("Cannot get count from DB: %v", err)
+		return err
+	}
+	if count == 0 {
+		log.Printf("Save TEST order cause DB is empty! \r\n")
+		if err := t.c.db.SaveOrderIntoDB(&database.OrderDb{
+			OrderID:         0,
+			Price:           0,
+			Hashrate:        0,
+			StartTime:       time.Time{},
+			ButterflyEffect: 2,
+			ActualStep:      start,
+		}); err != nil {
+			fmt.Printf("Cannot save order into DB %v\r\n", err)
+		}
+	}
+
+	pricePerMonthUSD, pricePerSecMh, err := t.GetPriceForTokenPerSec(token, symbol)
+	if err != nil {
+		log.Printf("Cannot get profit for tokens: %v\r\n", err)
+		return err
+	}
+	limitChargeInSNM := t.profit.LimitChargeSNM(balanceReply.GetSideBalance().Unwrap(), t.c.cfg.Sensitivity.PartCharge)
+	limitChargeInSNMClone := big.NewInt(0).Set(limitChargeInSNM)
+	limitChargeInUSD := t.profit.ConvertingToUSDBalance(limitChargeInSNMClone, snm.GetPrice())
+
+	mhashForToken, err := t.c.db.GetLastActualStepFromDb()
+	if err != nil {
+		log.Printf("Cannot get last actual step from DB %v\r\n", err)
+		return err
+	}
+
+	pricePackMhInUSDPerMonth := mhashForToken * (pricePerMonthUSD * t.c.cfg.Sensitivity.MarginAccounting)
+	sumOrdersPerMonth := limitChargeInUSD / pricePackMhInUSDPerMonth
+	if limitChargeInSNM.Int64() == 0 {
+		log.Printf("Balance SNM is not enough for create orders! %v", limitChargeInSNM.Int64())
+		return err
+	}
+
+	t.c.logger.Info("i", zap.String("CHARGE ORDERS ONCE: ", symbol),
```

Comment:
wow, so zap, such logger! Two lines above you're using `log` from stdlib, please be consistent.